### PR TITLE
boards,sys/arduino: major clean up

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

- Rename all `arduino_pinmap.h` to `arduino_iomap.h`
    - An empty `arduino_pinmap.h` that just includes `arduino_iomap.h` is provided for backward compatibility
    - Move all info from `arduino_board.h` into the new file as trivial macros, so that they can also be used outside of sketches
    - The new name reflects the fact not just pin mappings, but also other I/O features such as PWMs are mapped
- Drop all `arduino_board.h`
    - `arduino_board.h` and `arduino_iomap.h` now provide the exact same information, just in a different format
    - a generic `arduino_board.h` is provided instead that just uses the info in `arduinio_iomap.h` and provides them in the format the code in `sys/arduino` expects it
- Add fine grained features to indicate for mappings
    - availability of mappings for analog pins, DAC pins, PWM pins, UART devices, SPI/I2C buses to the corresponding RIOT identification can now be expressed:
        - `arduino_pins`: `ARDUINO_PIN_0` etc. are available
        - `arduino_analog`: `ARDUINO_A0` etc. are available
        - `arduino_pwm`: `ARDUINO_PIN_13_PWM_DEV` etc. are available
        - `arduino_dac`: `ARDUINO_DAC0` etc. are available
        - `arduino_uart`: `ARDUINO_UART_D0D1` or similar are available
        - `arduino_spi`: `ARDUINO_SPI_ISP` or similar are available
        - `arduino_i2c`: `ARDUINO_I2C_UNO` or similar are available
    - mechanical/electrical compatibility with specific form factors can now be expressed as features:
        - `aruino_shield_nano`: Arduino NANO compatible headers
        - `aruino_shield_uno`: Arduino UNO compatible headers
        - `aruino_shield_mega`: Arduino MEGA compatible headers
        - `aruino_shield_isp`: ISP header is available

This provides the groundwork to implement shield support as modules that can rely on the I/O mappings, rather than having to provide a configuration per board.

### Testing procedure

CI + lots of code review ...

In the long term the peripheral self testing shield will allow automatic testing for some of the I/O mappings for Arduino UNO compatible boards. But this PR would be required first for any soft of meaningful shield support in RIOT first.

### Issues/PRs references

None

### CODE REVIEW

**Comment by maribu** [comment]
@kfessel I split the `arduino_board.h` as you suggested out-of-band

File: doc/doxygen/src/kconfig/kconfig.md [hunk 1]
```diff
@@ -315,14 +315,14 @@ CPU family or vendor, it should be placed in the correspondent Kconfig file
 the Kconfig file of that module.
 
 #### Example
-The feature `arduino` is placed in `/kconfigs/Kconfig.features` and modelled
+The feature `arduino_pins` is placed in `/kconfigs/Kconfig.features` and modelled
 like:
 
 ```Kconfig
 config HAS_ARDUINO
```
Comment (kfessel) [inline doc/doxygen/src/kconfig/kconfig.md:322]:
```suggestion
config HAS_ARDUINO_PINS
```

**Review by kfessel — APPROVED** [review]
good cleanup PR:
- improves organisation
- readability
- completeness
maribu ran some tests, i trust his testing (no one will have all boards)
->

**Comment by maribu** [comment]
bors merge

**Comment by maribu** [comment]
Haven't seen that glitch in a while. I cannot say I missed it...

bors retry

**Comment by maribu** [comment]
bors retry

**Comment by maribu** [comment]
🎉  thx!

**Comment by gschorcht** [comment]
@maribu Was there any reason why the features `arduino_i2c` and `arduino_spi` were not added to `boards/common/nucleo144` with this PR?

**Comment by maribu** [comment]
Just to avoid growing the PRs scope. But if seems I never came back to adding those later on 😅

**Pull request merged**